### PR TITLE
Support multiple frames and tabs for magit fullscreen

### PR DESCRIPTION
**Pull request by pkryger**
When entering a fullscreen mode for `magit` commands a current window configuration is saved into a register. When the `magit` command is finished the configuration is restored from the register.

Instead of using a single register a new name is generated for each tab (when in `tab-bar-mode`) and each frame. That way, when there are multiple concurrent `magit` sessions in an Emacs instance each of them is holding to its own window configuration. With that, when a `magit` session is finished it restores a window configuration that had been in a given frame/tab.

I'm a little uncertain about register names. For one these are fairly long. Personally I'm not using `jump-to-register` a lot (almost none), but if someone does - this may become nuisance.

The second thing is a unique name for register. I decided to go with a frame address from core. It's extracted from a string representation (yuck 🤢), but it seems to support emacs-mac tabs (these from MacOS itself), as well as an Emacs client connected from iTerm.

As for testing: I've been using this for a last few weeks and haven't noticed any issues.

### CODE REVIEW

File: modules/init-git.el [hunk 1]
```diff
@@ -82,17 +85,28 @@
...
-      (cl-flet ((advice (orig-fun &rest args)
-                        (window-configuration-to-register :magit-fullscreen)
-                        (apply orig-fun args)
-                        (delete-other-windows)))
-        (advice-add symbol :around #'advice)))
-    (exordium-define-advice-magit-fullscreen 'magit-status)
-    (exordium-define-advice-magit-fullscreen 'exordium-magit-log)
-    (exordium-define-advice-magit-fullscreen 'magit-status-setup-buffer)
+    (defun exordium--magit-fullscreen (orig-fun &rest args)
+      (let* ((frame-address
```
Comment (philippe-grenet) [inline modules/init-git.el:89]: That's cute, but if you only need to generate a unique name, did you consider using `gensym` (or its variant `cl-gensym`)? This is typically what you used in Lisp, especially while defining macros where you want to make sure that whatever local variables or functions your macro will include in the generated code have unique names with no risk of clashing with the code where the macro happens to be used.

**Review by philippe-grenet — APPROVED** [review]
One small comment for your consideration. If not applicable, feel free to merge as is.

File: modules/init-git.el (hunk 1 above)
Comment (pkryger) [reply modules/init-git.el:89]: Thanks for that! Wasn't aware of that functionality and the macro idiom. `gensym` it is.

Comment (pkryger) [reply modules/init-git.el:89]: Second thought: that would mean that there's a new register for each call of the advice. So, effectively for each invocation of  `magit-status` et al. I think I my call count of these functions is in 3 digits per session. On the other hand, I tend to stay in a one frame and use single digit number of tabs. I could cash that, but that would mean that I need some key for each frame/tabbar-tab and an extra variable. Yes - register names will be short, but do you think it's worth it?

**Review by philippe-grenet — APPROVED** [review]

File: modules/init-git.el (hunk 1 above)
Comment (philippe-grenet) [reply modules/init-git.el:89]: my mistake, following discussion your design is correct.

**Pull request merged**